### DISCUSSION TITLE
OSError: [Errno 1] EPERM

### DISCUSSION THREAD

**Discussion by Vikashkushwaha70** [micropython/micropython#15680] [General]
import time
import network
from machine import Pin
import BlynkLib

# Define pins
led1 = Pin(5, Pin.OUT)
led2 = Pin(2, Pin.OUT)  # Uncommented for proper setup

# Wi-Fi credentials
ssid = 'Vikash'
password = '123123178'

# Connect to Wi-Fi
sta_if = network.WLAN(network.STA_IF)
sta_if.active(True)
if not sta_if.isconnected():
    print('Connecting to network...')
    sta_if.connect(ssid, password)
    while not sta_if.isconnected():
        time.sleep(1)
print('Network config:', sta_if.ifconfig())

# Blynk authentication token
BLYNK_AUTH = 'fhMTP8zz6x1Oc6cf3kENlUDmkC8b6E2D'

# Initialize Blynk
blynk = BlynkLib.Blynk(BLYNK_AUTH)

# Register virtual pin handler for V0
@blynk.on("V0")
def v0_write_handler(value):
    if int(value[0]) == 1:
        led1.value(1)  # Turn the LED on
    else:
        led1.value(0)  # Turn the LED off

# Register virtual pin handler for V1
@blynk.on("V1")
def v1_write_handler(value):
    if int(value[0]) == 1:
        led2.value(1)  # Turn the LED on
    else:
        led2.value(0)  # Turn the LED off

# Main loop
while True:
    blynk.run()

Connecting to network...
Traceback (most recent call last):
  File "<stdin>", line 19, in <module>
OSError: [Errno 1] EPERM

**Comment by Josverl** [comment]
You can use [GitHub Flavoured Markdown](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/about-writing-and-formatting-on-github) to format your posts. In particular use ```py ... ``` around your code snippets to get syntax highlighting, see the section below.

https://github.com/orgs/micropython/discussions/9111

**Comment by shariltumin** [comment]
Which board are you using to run your program? Are you sure that your board supports WiFi?

**Reply by Vikashkushwaha70** [reply]
Yes, my board has WiFi, but I checked, and the WiFi function is not working. Thank you for your time.
